client/components/voip/SimpleVoipUser.ts

```diff
@@ -0,0 +1,225 @@
+import { Emitter } from '@rocket.chat/emitter';
+
+import { ClientLogger } from '../../../lib/ClientLogger';
+import { CallState } from './Callstate';
+import { ICallerInfo, ICallEventDelegate } from './ICallEventDelegate';
+import { IConnectionDelegate } from './IConnectionDelegate';
+import { IRegisterHandlerDelegate } from './IRegisterHandlerDelegate';
+import { VoIPUser } from './VoIPUser';
+import { IMediaStreamRenderer, VoIPUserConfiguration } from './VoIPUserConfiguration';
+
+interface IState {
+	isReady: boolean;
+	enableVideo: boolean;
+}
+
+export enum CallType {
+	AUDIO,
+	AUDIO_VIDEO,
+}
+
+export enum VoipEvents {
+	'initialised',
+	'connected',
+	'connectionerror',
+	'registered',
+	'registrationerror',
+	'unregistered',
+	'unregistrationerror',
+	'incomingcall',
+	'callestablished',
+	'callterminated',
+	'negotiationfailed',
+}
+export class SimpleVoipUser
+	implements IRegisterHandlerDelegate, IConnectionDelegate, ICallEventDelegate
+{
+	state: IState;
+
+	userHandler: VoIPUser;
+
+	userName: string;
+
+	password: string;
+
+	registrar: string;
```

Comment:
Portuguese? `registrar: string;` 🤔

Reply:
No it is english. It is an entity in sip world which keeps the record of logged in endpoints. endpoints communicate with registrar to do registration.